### PR TITLE
fix(runtime): only generate lazy build CSS when there are component tags

### PR DESCRIPTION
**Pull request by tanner-reits**
## What is the current behavior?

If a developer accidentally calls `defineCustomElements()` multiple times with the same component metadata, invalid CSS will be generated and injected into the DOM. This happens because we only create a `cmpTags` array based on new component tags added to the Custom Element registry, but we do not check if the array has a length before appending additional CSS rules to the tag selectors.

Closes: #3771

## What is the new behavior?

This commit fixes an issue where invalid CSS could be generated and inserted into the DOM if `bootstrapLazy` was called with no component metadata or if no new components are registered in the custom element registry (i.e. it is called multiple times with the same metadata).

## Documentation

N/A

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Testing

**Automated testing**
Added some new unit tests to `bootstrap-lazy.spec.tsx`

**Manual testing**
This can be tested using the reproduction [in this comment](https://github.com/ionic-team/stencil/pull/3771#issuecomment-1300036253) from #3771 and a build of this branch.

**Note:** When testing I was experiencing some caching issues with the wrong version of the runtime being used so you may need to nuke `node_modules` and lock files when installing this build into the project

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
👍

**Comment by tanner-reits** [comment]
@rwaskiewicz Removed those tests as discussed

File: src/runtime/bootstrap-lazy.ts [hunk 1]
```diff
@@ -196,29 +196,32 @@ export const bootstrapLazy = (lazyBundles: d.LazyBundlesRuntimeData, options: d.
...
+    }
 
-  // Add hydration styles
-  if (BUILD.invisiblePrehydration && (BUILD.hydratedClass || BUILD.hydratedAttribute)) {
-    dataStyles.innerHTML += cmpTags + HYDRATED_CSS;
-  }
+    // Add hydration styles
+    if (BUILD.invisiblePrehydration && (BUILD.hydratedClass || BUILD.hydratedAttribute)) {
+      dataStyles.innerHTML += cmpTags + HYDRATED_CSS;
+    }
```
Comment (rwaskiewicz) [inline src/runtime/bootstrap-lazy.ts:210]: Non-blocking - Can we add a TODO to get this part under unit test?

**Comment by rwaskiewicz** [comment]
@tanner-reits I found a few additional files for the invisible prehydration testing that I removed in cdafaa0 - can you & @christian-bromann PTAL? Otherwise, this looks good to me, I had just one non-blocking comment

**Review by rwaskiewicz — APPROVED** [review]

**Comment by tanner-reits** [comment]
@rwaskiewicz Thanks for that, LGTM. Wasn't sure if we'd wanna bring these tests back if the ability to remove an element from the registry became a thing, but I suppose we can revert all those changes if that time comes.

**Review by christian-bromann — APPROVED** [review]

**Pull request merged**